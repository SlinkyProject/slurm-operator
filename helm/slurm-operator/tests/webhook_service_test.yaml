# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
---
suite: test webhook service
templates:
- webhook/service.yaml
chart:
  version: 1.2.3
  appVersion: 1.2.3
release:
  name: test-release
  namespace: test-namespace

tests:
- it: manifest should match snapshot
  asserts:
  - matchSnapshot: {}

- it: should not render when disabled
  set:
    webhook:
      enabled: false
  asserts:
  - hasDocuments:
      count: 0

- it: should set namespace
  set:
    namespaceOverride: custom-namespace
  asserts:
  - equal:
      path: metadata.namespace
      value: custom-namespace


- it: should set name
  set:
    nameOverride: custom-name
  asserts:
  - equal:
      path: metadata.name
      value: custom-name-webhook
