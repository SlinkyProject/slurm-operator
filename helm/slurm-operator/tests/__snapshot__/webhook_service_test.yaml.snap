manifest should match snapshot:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: slurm-operator-webhook
        app.kubernetes.io/version: 1.2.3
        helm.sh/chart: slurm-operator-1.2.3
      name: slurm-operator-webhook
      namespace: test-namespace
    spec:
      ports:
        - name: https
          port: 443
          protocol: TCP
          targetPort: 9443
        - name: health
          port: 8081
          protocol: TCP
          targetPort: 8081
      selector:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/name: slurm-operator-webhook
      type: ClusterIP
